### PR TITLE
Use errors.Is for error handling

### PR DESCRIPTION
**Pull request by alexander-melentyev**

### CODE REVIEW

File: transport/http/jsonrpc/client_test.go [hunk 1]
```diff
@@ -170,7 +171,7 @@ func TestClientHappyPath(t *testing.T) {
 		}
 
 		b, err := ioutil.ReadAll(r.Body)
-		if err != nil && err != io.EOF {
```
Comment (sagikazarmark) [inline transport/http/jsonrpc/client_test.go:173]: See https://mobile.twitter.com/mvdan_/status/1442492461025222661

File: metrics/cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -271,10 +271,6 @@ func sum(a []float64) float64 {
 	return v
 }
 
-func last(a []float64) float64 {
-	return a[len(a)-1]
-}
-
```
Comment (sagikazarmark) [inline metrics/cloudwatch/cloudwatch.go:277]: This should probably be a separate commit/PR.

File: transport/http/jsonrpc/client_test.go (hunk 1 above)
Comment (alexander-melentyev) [reply transport/http/jsonrpc/client_test.go]: Added back

File: metrics/cloudwatch/cloudwatch.go (hunk 2 above)
Comment (alexander-melentyev) [reply metrics/cloudwatch/cloudwatch.go]: Done

**Review by sagikazarmark — APPROVED** [review]

**Comment by peterbourgon** [comment]
I guess we should stick sd/eureka on the deprecation list...

**Comment by alexander-melentyev** [comment]
Again...
